### ISSUE TITLE
Fatal FIPS selftest failure importing from sentence_transformers (RHEL8 FIPS enabled failure)

### ISSUE THREAD

**Issue by apetersonflc** [UKPLab/sentence-transformers#2234]
I am working with a RHEL8 box with FIPS enabled and importing sentence_transformers causes a FATAL FIPS SELFTEST FAILURE:

>>> from sentence_transformers import SentenceTransformer
fips.c(145): OpenSSL internal error, assertion failed: FATAL FIPS SELFTEST FAILURE
Aborted (core dumped)

From what I understand, a package created on a machine with FIPS disabled may not work on a machine with FIPS enabled.

Any work arounds or ideas?

Followup: Everything works with FIPS disabled so wanted to share that.  Unfortuntately, FIPS disabled is not an option for my prod environments.

**Comment by jayteaftw** [comment]
Hi did you ever fix this. I havent had this issue on a FIPS machine until today

**Comment by apetersonflc** [comment]
Yes, but it isn't a straightforward process.  Long story short, you need to remove md5 as a crypto algorithm by setting usedforsecurity=False anywhere you can.  The issue is FIPS wont let you use md5 so it must be disabled.

<img width="1084" height="474" alt="Image" src="https://github.com/user-attachments/assets/5f6db774-21bd-4ade-b68b-8dae4bc5c75e" />

**Comment by jayteaftw** [comment]
Hi thanks for reaching out! In my code! However, I dont have any instance of hashlib.md5 being initialized. 
Its occurring when I import the package

```python
   if embedding_model and engine == "":
        print("Check in get_if: Before import SentenceTransformer")
        from sentence_transformers import SentenceTransformer #<------ Fails here!
        print("Check in get_if: After import SentenceTransformer")
        try:
            ef = SentenceTransformer(
                get_model_path(embedding_model, auto_update),
                device=DEVICE_TYPE,
                trust_remote_code=RAG_EMBEDDING_MODEL_TRUST_REMOTE_CODE,
                backend=SENTENCE_TRANSFORMERS_BACKEND,
                model_kwargs=SENTENCE_TRANSFORMERS_MODEL_KWARGS,
            )
        except Exception as e:
            log.debug(f"Error loading SentenceTransformer: {e}")

    return ef
```

**Comment by kobaan** [comment]
The non-FIPS-compliant md5 algorithm from hashlib was introduced here:
https://github.com/UKPLab/sentence-transformers/pull/3338/files

Would be better to use FIPS compliant SHA-2 instead of MD5

**Comment by tomaarsen** [comment]
@kobaan well spotted, thanks! Will switch to SHA-256 with `usedforsecurity=False` in a new PR. I'd happily welcome anyone with FIPS-compliance testing the PR branch.

- Tom Aarsen
